src/Icicle/Avalanche/Check.hs

```diff
@@ -88,7 +88,13 @@ checkStatement frag ctx stmt
                    $ checkExp frag (ctxExp ctx) x
                Let n x' <$> go stmts
 
-        ForeachInts n from to stmts
+        While t n to stmts
+         -> do to'   <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) to
+
+               While t n to' <$> go stmts
```

Comment:
I guess this should get `n` from the accumulator map and check that `to'` has the same type